src/invert/laplace/impls/multigrid/multigrid_laplace.cxx

```diff
@@ -417,18 +417,17 @@ BOUT_OMP(for)
   // Compute solution.
 
   mgcount++;
-  if((mgcount == 300) && (pcheck == 0)) {
-    tcheck = 1;
-  }
-  t0 = MPI_Wtime();
+  if (pcheck > 0) t0 = MPI_Wtime();
 
   kMG->getSolution(std::begin(x), std::begin(b), 0);
 
-  t1 = MPI_Wtime();
-  if((mgcount == 300) && (tcheck != pcheck)) tcheck = pcheck;
-  soltime += t1-t0;
-  if(mgcount%300 == 0) {
-    output<<"Accumulated execution time at "<<mgcount<<" Sol "<<soltime<<" ( "<<settime<<" )"<<endl;
+  if (pcheck > 0) {
+    t1 = MPI_Wtime();
+    if((mgcount == 300) && (tcheck != pcheck)) tcheck = pcheck;
```

Comment:
What does tcheck do? It doesn't seem to be used in this file.

Reply:
As far as I can see it was not used. Don't know what the original intention was. I have removed it, and also an 'rcheck', in a new commit.